addons/sys_data/script_component.hpp

```diff
@@ -14,4 +14,6 @@
     #define DEBUG_SETTINGS DEBUG_SETTINGS_SYS_DATA
 #endif
 
+#define IS_SERIALIZEDHASH(array) (IS_ARRAY(array) && {(count array) > 0} && {IS_STRING((array select 0))} && {(array select 0) == "ACRE_HASH"})
```

Comment:
`arrayIsEqualTo []` will return `false` if it's not empty and if it's not array, so it guarantees it is array. Yes it will be case-sensitive, unsure if it matters, can just add `toLower` if it does.

I am not sure where all this is used, that's why I said separate PR. It's very minor optimization if any at all anyways.